### DISCUSSION TITLE
ESP32-S3 and parallel display: useable pins?

### DISCUSSION THREAD

**Discussion by jho59** [Bodmer/TFT_eSPI#3243] [General]
Hi,

I like to use a BananaPi Leaf S3 (it's a ESP32-S3 board) and display with parallel port.
I wonder about the comment in a header file:
```
// Must use pins in the range 0-31 for the data bus
// so a single register write sets/clears all bits
```
The board has 8 pins side by side, so it would be easy to connect, but numbers are GPIO35 - GPIO42.  Can I use these pins?

**Comment by Bodmer** [comment]
For the S3, higher GPIO can be used. See example setup here:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup207_LilyGo_T_HMI.h

**Reply by jho59** [reply]
Thanks, but it doesn't work.
I want to use a 7" display with SSD1963 controller with an ESP32-S3 (BPI Leaf S3).
My simple program is:

```
#include <TFT_eSPI.h>       // Include the graphics library
TFT_eSPI tft = TFT_eSPI();  // Create object "tft"

void setup(void) {
  Serial.begin(115200);
  tft.init();
  delay(2500);
  tft.fillScreen(TFT_GREEN);
}

void loop() { }
```


And connections are:
```
#define TFT_CS     4 
#define TFT_DC     5 	// RS at display 
#define TFT_RST    7 

#define TFT_WR     6 
#define TFT_RD    -1  

#define TFT_D0   35 
#define TFT_D1   36 
#define TFT_D2   37 
#define TFT_D3   38 
#define TFT_D4   39 
#define TFT_D5   40 
#define TFT_D6   41 
#define TFT_D7   42 
```

After tft.init() I see "pixel noise" on display for 2,5 sec, then display gets dark.
I changed connections to left side of board, so pins are lower then 31. I can see a green display. Examples like "ColourWheel" or "Julia fractals" are working fine.

Again connections to higher pins and only pixel noise.
So I think, I don't make a mistake with wiring.

**Comment by bionicbone** [comment]
The Setup207 example did not work for me either...  
I suspect it has something to do with Bomer's comment "single register write sets/clears all bits"
I did try looking at the ESP32S3 Tech Ref Manual to see if I could the good answer but more confused :)
In the end I just did this and it is working...

// Base Defines
#define USER_SETUP_ID 50
#define TFT_PARALLEL_8_BIT
#define SSD1963_800BD_DRIVER    // 480 x 800 display sourced from https://www.buydisplay.com/7-tft-screen-touch-lcd-display-module-w-ssd1963-controller-board-mcu
#define TFT_RGB_ORDER TFT_BGR  // Colour order Blue-Green-Red

// ESP32S3 pins used  (aliexpress Dev Board N16R8)
#define TFT_CS   -1 // tied to GND, NR for parallel  // Chip select control pin
#define TFT_DC   3  // Data Command control pin - must use a pin in the range 0-31
#define TFT_RST  -1 // tied to ESP32S3 RST Pin (also TFT_eSPI has software reset)  // Reset pin
#define TFT_WR   8 // Write strobe control pin - must use a pin in the range 0-31
#define TFT_RD   -1 // Display read not used in my project, but display can be read so may need a pin later

#define TFT_D0   4 // Must use pins in the range 0-31 or alternatively 32-48
#define TFT_D1   5 // so a single register write sets/clears all bits.
#define TFT_D2   6 // Pins can be randomly assigned, this does not affect
#define TFT_D3   7 // TFT screen update performance.
#define TFT_D4   15
#define TFT_D5   16
#define TFT_D6   17
#define TFT_D7   18
